### PR TITLE
Create a concrete model for Jira Server

### PR DESCRIPTION
**Pull request by ChomickiM**
Small commit to create the model for all of the other Jira Server changes to come. Mostly adding you guys to this PR for awareness as well as checking if anything is missing or should be added. Since the "Install Plugin Remotely" is its own action, I thought it doesn't really fit to add anything for it into the model.

### CODE REVIEW

File: channel-jira-server/src/main/java/com/synopsys/integration/alert/channel/jira/server/model/JiraServerGlobalConfigModel.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+        jiraServerGlobalConfigModel.setUserName(userName);
+        jiraServerGlobalConfigModel.setDisablePluginCheck(disablePluginCheck);
+
+        jiraServerGlobalConfigModel.setIsPasswordSet(StringUtils.isNotBlank(password));
+        jiraServerGlobalConfigModel.setPassword(null);
+
+        return jiraServerGlobalConfigModel;
+    }
+
+    public Optional<String> getUrl() {
```
Comment (bamandel) [inline channel-jira-server/src/main/java/com/synopsys/integration/alert/channel/jira/server/model/JiraServerGlobalConfigModel.java:36]: How do we feel about this optional approach for all our fields? I'm not a fan of a required field potentially being missing. I think this approach may hide other potential issues in the code base

Comment (bamandel) [reply channel-jira-server/src/main/java/com/synopsys/integration/alert/channel/jira/server/model/JiraServerGlobalConfigModel.java:36]: I understand that we did it this way for email but want to be sure we're happy with the decision

Comment (ChomickiM) [reply channel-jira-server/src/main/java/com/synopsys/integration/alert/channel/jira/server/model/JiraServerGlobalConfigModel.java:36]: If we removed the optionals, wouldn't places like the accessors require its own validation? Right now those are validated as part of the helper classes before the accessors are called. These models don't have constructors and are created field by field using the setters so wouldn't we run the risk of potential NPEs without the optionals?

Comment (psantos1113) [reply channel-jira-server/src/main/java/com/synopsys/integration/alert/channel/jira/server/model/JiraServerGlobalConfigModel.java]: We will need to add tests to the action class in the future to validate that null values aren't allowed going forward.  We will need to change email, proxy, and encryption related classes in the future as well.

**Review by psantos1113 — APPROVED** [review]
